marahgh/assignment4/IslandCounter.cpp

```diff
@@ -0,0 +1,50 @@
+#include <iostream>
+#include <vector>
+#include "IslandCounter.h"
+
+using namespace std;
+
+/*returns true if the given coordinates ar in the grid's bounds, flase
+ * otherwise*/
+bool IslandCounter::inBounds(int i, int j) {
+  return (i >= 0 && i < grid.size()) && (j >= 0 && j < grid[i].size());
+}
+
+/*return true if the given coordinates are of  a legal neighbor,
+(up/down/left/right - this happens if |deltaI-deltaJ|=1)
+ and false otherwise*/
+bool IslandCounter::isLegalNeighbor(int i, int j, int deltaI, int deltaJ) {
+  return (inBounds(i, j) && (abs(deltaI - deltaJ) == 1));
+}
+
+/*updates the land neighbors cells which belong to the current island*/
+void IslandCounter::updateNeighbors(int i, int j) {
+  for (int rowDelta : {0, 1, -1}) {
+    for (int colDelta : {0, 1, -1}) {
+      int newI = i + rowDelta;
+      int newJ = j + colDelta;
+      if (isLegalNeighbor(newI, newJ, rowDelta, colDelta) && grid[newI][newJ]) {
+        grid[newI][newJ] = false;

```

Comment:
Minor detail, but if you move `grid[newI][newJ] = false;` to the beginning of the function as `grid[i][j] = false;` you don't need to do that in `countIslands()`